src/lib/utils/npm-user-agent-parser.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Parses a user agent string for NPM metadata.
+ * @param {string} userAgent 
+ * @return {{npmVersion: string, nodeVersion: string, platform: string, architecture: string}}
+ */
+const npmUserAgentParser = (userAgent) => {
```

Comment:
It might be easier for you to move it out once it's merged as I don't have the ability to create new repositories against the verdaccio account.